ui/app/AppLayouts/Wallet/stores/RootStore.qml

```diff
@@ -177,6 +178,10 @@ QtObject {
 //            walletModelV2Inst.collectiblesView.collections.getCollectionTraitMaxValue(collectionIndex, traitType).toString();
     }
 
+    function selectCollectible(slug, id) {
```

Comment:
I feel like this call should go to the collections module and we internally update the currentCollectible which is always used in the detail view?
I feel it would be cleaner but if you prefer it this way, let me know

Reply:
also more so cause I see that the current_collectible controller has the instance to the collectionsModule and that not the way we communicate between modules and submodules currently.

We could have the current_collectible module instantiated in the main collectibles modules and tackle all of this there

Reply:
Just to clarify, we currently have:

```
main
|-collectible (Only contains the name of the collectibles, I think this one is not used anywhere so I'm not sure what the original intention for this was. Could be removed in this PR or I could do it in a separate one)
|-collectibles
|-collections
|-current_collectible
```

Do you suggest having `current_collectible` instanced under `collectibles`? My main reasoning for doing it like I did is that the "current_collectible" needs information from both `collections` (image, name) and `collectibles` (everything else), so I figured it needed to be separate from both of them. The `current_collectible/controller` gets the data from `collectibles` and `collections` instead of some service because that's the only place where the data is stored after fetching it, but perhaps storing references to the other modules is not the best option. Is there other path I should use to access those pieces of information?

Ideally, I think we could move to a tree-like data structure where the main module has a list of collections, and each collection contains a list collectibles. That way, retrieving the information we need for the details would be more straightforward (collectible+parent).

Reply:
Hey you are right, I just dug deeper and I can see that even the way the collectibles are currently updated is not correct, ideally it should have been via the collectibles and like you said yes it should have been a tree structure. so I guess we can do this later. Just create another issue for it, so that the name and image we need from collections is updated to collectibles already and then we only find that item and update it in the current_collectible.

Approving for now :)